tools/cmake/tvos.toolchain.cmake

```diff
@@ -8,10 +8,15 @@ set_common_xcode_attributes()
 
 set(REALM_SKIP_SHARED_LIB ON)
 
+# CMake special-cases the compiler detection for iOS in a way that happens
+# to also work for tvOS, so use the iphoneos base SDK to opt-in to that and
+# then override the platform to tvOS below
+set(CMAKE_OSX_SYSROOT "iphoneos")
```

Comment:
AFAICT it's not so much a bug as that CMake just doesn't actually support tvOS/watchOS.